v2/cf-nodejs-logging-support/src/lib/config/config.ts

```diff
@@ -129,6 +134,36 @@ export default class Config {
             }
 
             file.fields?.forEach(field => {
+                if (field.envVarSwitch) {
+                    let isEnabled = isEnvVarEnabled(field.envVarSwitch);
+                    if (isEnabled) {
+                        field._meta = {
+                            isRedacted: false,
+                            isEnabled: true,
+                        }
+                    } else {
+                        field._meta = {
+                            isRedacted: false,
+                            isEnabled: false,
+                        }
+                    }
+                }
+
+                if (field.envVarRedact) {
+                    let isEnabled = isEnvVarEnabled(field.envVarRedact);
+                    if (isEnabled) {
+                        field._meta = {
+                            isRedacted: true,
+                            isEnabled: true,
+                        }
+                    } else {
+                        field._meta = {
+                            isRedacted: true,
+                            isEnabled: false,
+                        }
+                    }
+                }
```

Comment:
So `_meta` is optional? I would not do that, it encodes more possible states than necessary and makes the logic more complicated.

In my opinion we should simply define the effect of isEnabled and isRedacted as follows:

* `isEnabled?`
  * `false`: omit the field
  * `true`:
    * `isRedacted?`
      * `false`: write actual value or omit if empty 
      * `true`: write `redacted` or omit if empty

Both fields should be present in `_meta` for every field. We can determine them as follows:

```js
if (field.envVarSwitch) {
  _meta.isEnabled = isEnvVarEnabled(field.envVarSwitch)
} else {
  _meta.isEnabled = true
}

if (field.envVarRedact) {
  _meta.isRedacted = !isEnvVarEnabled(field.envVarRedact) // if the env var is actually set to true, we do not redact => invert result
} else {
  _meta.isRedacted = false
}
```

Btw, I used the term "omit" here, because it will be the default behavior. However, if a field is mandatory and has a default value, we use it here instead of omitting.